[2023-06-13T12:16:40.530+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2023-06-13T12:06:25.188291+00:00 [queued]>
[2023-06-13T12:16:40.536+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2023-06-13T12:06:25.188291+00:00 [queued]>
[2023-06-13T12:16:40.536+0000] {taskinstance.py:1308} INFO - Starting attempt 3 of 4
[2023-06-13T12:16:40.547+0000] {taskinstance.py:1327} INFO - Executing <Task(BashOperator): pyspark_train_classifier_model> on 2023-06-13 12:06:25.188291+00:00
[2023-06-13T12:16:40.551+0000] {standard_task_runner.py:57} INFO - Started process 288 to run task
[2023-06-13T12:16:40.558+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'agile_data_science_batch_prediction_model_training', 'pyspark_train_classifier_model', 'manual__2023-06-13T12:06:25.188291+00:00', '--job-id', '15', '--raw', '--subdir', 'DAGS_FOLDER/setup.py', '--cfg-path', '/tmp/tmpo8l8mitx']
[2023-06-13T12:16:40.560+0000] {standard_task_runner.py:85} INFO - Job 15: Subtask pyspark_train_classifier_model
[2023-06-13T12:16:40.611+0000] {task_command.py:410} INFO - Running <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2023-06-13T12:06:25.188291+00:00 [running]> on host 216411ef4acd
[2023-06-13T12:16:40.677+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='agile_data_science_batch_prediction_model_training' AIRFLOW_CTX_TASK_ID='pyspark_train_classifier_model' AIRFLOW_CTX_EXECUTION_DATE='2023-06-13T12:06:25.188291+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-06-13T12:06:25.188291+00:00'
[2023-06-13T12:16:40.682+0000] {subprocess.py:63} INFO - Tmp dir root location: 
 /tmp
[2023-06-13T12:16:40.683+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', '\nspark-submit --master spark://spark-master-train:7077   /Users/admin/Documents/test-ml/fiware-mlops-supermarket///resources/train_spark_mllib_model.py   /Users/admin/Documents/test-ml/fiware-mlops-supermarket//']
[2023-06-13T12:16:40.692+0000] {subprocess.py:86} INFO - Output:
[2023-06-13T12:16:40.697+0000] {subprocess.py:93} INFO - /bin/bash: line 2: spark-submit: command not found
[2023-06-13T12:16:40.698+0000] {subprocess.py:97} INFO - Command exited with return code 127
[2023-06-13T12:16:40.710+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/bash.py", line 211, in execute
    f"Bash command failed. The command returned a non-zero exit code {result.exit_code}."
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 127.
[2023-06-13T12:16:40.725+0000] {taskinstance.py:1350} INFO - Marking task as UP_FOR_RETRY. dag_id=agile_data_science_batch_prediction_model_training, task_id=pyspark_train_classifier_model, execution_date=20230613T120625, start_date=20230613T121640, end_date=20230613T121640
[2023-06-13T12:16:40.741+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 15 for task pyspark_train_classifier_model (Bash command failed. The command returned a non-zero exit code 127.; 288)
[2023-06-13T12:16:40.766+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-06-13T12:16:40.797+0000] {taskinstance.py:2651} INFO - 0 downstream tasks scheduled from follow-on schedule check
